astro/src/content/docs/extend/events-and-webhooks/events/user-login-failed.mdx

```diff
@@ -47,6 +47,22 @@ export const eventType = 'user.login.failed';
     Moved to <InlineField>event.info</InlineField> in 1.30.0
   </APIField>
 
+  <APIField slot="trailing-fields" name="event.reason" type="Object" since="1.53.0">
+    An object containing data on the reason the login failed if it was prevented by a [Login Validation Lambda](/docs/extend/code/lambdas/login-validation).
+  </APIField>
+
+  <APIField slot="trailing-fields" name="event.reason.code" type="String" since="1.53.0">
+    The theme message key used to find a message to display to the end user.
```

Comment:
This is the cause of the event. Prior to this version, the only time we sent this event was due to bad credentials. Now that may send this event even when your credentials are correct, there is a reason code field. 

This can be `credentials` or `lambdaValidation`.